packages/eservice-descriptors-archiver/src/services/archiveDescriptorProcessor.ts

```diff
@@ -0,0 +1,98 @@
+import { Logger } from "pagopa-interop-commons";
+import {
+  Agreement,
+  DescriptorId,
+  EServiceId,
+  agreementState,
+  descriptorState,
+  genericInternalError,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import { ReadModelService } from "./readModelService.js";
+
+export async function archiveDescriptorForArchivedAgreement(
```

Comment:
Keeping logic confined in this processor file so I can test it easily.
The reference logic in Scala is here: https://github.com/pagopa/interop-be-jobs/blob/1.0.x/eservice-descriptors-archiver/src/main/scala/it/pagopa/interop/eservicedescriptorsarchiver/util/JobExecution.scala